students/ChrisH/lesson10/mailroom.py

```diff
@@ -0,0 +1,335 @@
+        print(rpt_title)
+        print("-" * len(rpt_title))
+        for donor in self.donorlist:
+            print(f"{donor.name:{name_max}}  ", end='')
+            ddons = donor.donations
+            print(f"$ {sum(ddons):>10.2f}   {len(ddons):>9}  ${sum(ddons)/len(ddons):>12.2f}")
+
+    def send_letters_all(self):
+        """
+            Runs through donor data structure, generates a thank you letter for each and saves it to
+            the current working directory with in a date+donorname.txt file
+            :return: None
+            """
+        for donor in self.donorlist:
+            print(f'Generating letter for {donor.name}')
+            now = time.localtime()
+            f_name = f"{now.tm_year}{now.tm_mon:02}{now.tm_mday:02}_"
+            f_name += donor.name.replace(" ", "_") + ".txt"
+            with open(f_name, 'w') as file_out:
+                file_out.write(donor.generate_letter())
+        return None
+
+    @property
+    def list_donors(self):
+        return [donor.name for donor in self.donorlist]
+
+    @property
+    def count(self):
+        return len(self.donorlist)
+
+    def get_donor(self, dname):
+        if dname == "":
+            return None
+
+        for donor in self.donorlist:
+            if donor.name == dname:
+                return donor
+
+        newdonor = Donor(dname)
+        self.add_donor(newdonor)
+        return newdonor
+
+    def load_donorlist(self):
+        with open(self.DATA_FILE, 'rb') as file_in:
+            self.donorlist = pickle.load(file_in)
+        return self.count
+
+    def save_donorlist(self):
+        with open(self.DATA_FILE, 'wb') as file_out:
+            pickle.dump(self.donorlist, file_out)
+        return self.count
+
+    def total_contribution(self):
+        return sum([sum(d.donations) for d in self.donorlist])
+
+
+    def challenge(self, mul, min_don=None, max_don=None):
+        """
+        Multiplies donations, returns a Donors object.
+        Note that this function destroys the original order of the donations, as multiplied donation values get
+        grouped together, and non-multiplied donations get moved to the end of the list.
+        :param mul: a multiplier to be applied to each donation value of each donor
+        :param min_don: multiplier only applies to values greater than this amount, if None - does not apply
+        :param max_don: multiplier only applies to values less than or equal to this amount, if None - does not apply
+        :return: a new Donors object, with multiplied donation values
+        """
+
+        # define the filter functions, based on if min or max values for donation multipliers were added
+        # unfortunately, PEP8 abhors assigning lambdas to variables, so here are some glorious Defs.
+        #  fi = donations that should be multiplied
+        #  not_fi = donations that should not be multiplied
+        if min_don and max_don:
+            def fi(x): return max_don >= x > min_don
```

Comment:
just make the variable name `donations_to_be_multiplied` and `donations_not_to_be_modified`